File: examples/SDK/golang/WorkingWithTables/deleteGlobalTableRegion/deleteGlobalTableRegion.go
package main

import (
    "github.com/aws/aws-sdk-go/aws"
    "github.com/aws/aws-sdk-go/aws/session"
    "github.com/aws/aws-sdk-go/service/dynamodb"
    "fmt"
)

var table     = "Music"
var awsRegion = "ap-northeast-2"

func getSession() (*session.Session) {
    sess := session.Must(session.NewSessionWithOptions(session.Options{
        SharedConfigState: session.SharedConfigEnable,
        // Provide SDK Config options, such as Region and Endpoint
        Config: aws.Config{
            Region: aws.String(awsRegion),
        },
    }))

    return sess
}

func updateTable() error {
    dynamoDBClient := dynamodb.New(getSession())

    replicaUpdates := []*dynamodb.ReplicaUpdate{
        {
            Delete: &dynamodb.DeleteReplicaAction{
                RegionName: aws.String(awsRegion),
            },
        },
    }

    _, err := dynamoDBClient.UpdateGlobalTable(&dynamodb.UpdateGlobalTableInput{
        GlobalTableName: &table,
        ReplicaUpdates: replicaUpdates,
    })

    if err != nil {
        fmt.Println("Got error updating Global Table.", err)
        return err
    }

    return nil
}

func main() {
    fmt.Println("Creating Global Table ...")
    updateTable()
    fmt.Println("Finished ...")
}

File: examples/SDK/node.js/sdk_v3/control_plane/WorkingWithTables/delete-global-table-region.js
const { DynamoDBClient, UpdateTableCommand } = require("@aws-sdk/client-dynamodb");

const dynamodb = new DynamoDBClient({ region: "us-west-2" });

const tableName = "Music";

async function deleteGlobalTableRegion() {
  const params = {
    TableName: tableName,
    ReplicaUpdates: [
      {
        Delete: {
          RegionName: "ap-northeast-2",
        },
      },
    ],
  };

  const command = new UpdateTableCommand(params);
  const response = await dynamodb.send(command);
  return response;
}

deleteGlobalTableRegion()
  .then((data) => console.log(JSON.stringify(data, null, 2)))
  .catch((error) => console.error(JSON.stringify(error, null, 2)));
